src/nvim/linematch.c

```diff
@@ -331,6 +331,13 @@ size_t linematch_nbuffers(const char **diff_blk, const int *diff_len, const size
                           int **decisions, bool iwhite)
 {
   assert(ndiffs <= LN_MAX_BUFS);
+  // check for valid inputs
+  for (size_t i = 0; i < ndiffs; i++) {
+    if (diff_blk[i] == NULL) {
```

Comment:
This is a very aggressive bandaid. Ideally we should figure how `diff_blk` is being called with NULL pointers since that's where the real bug lies.